v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/transforms/SourceWriterFn.java

```diff
@@ -291,42 +286,43 @@ public void processElement(ProcessContext c) {
+        String message = ex.getMessage();
+        if (cause != null) {
+          message += ", Caused by: " + cause.getMessage();
+        }
+
         if (cause == null) {
+          // Log all retryable errors. It will be helpful for debugging since the files in retryable
+          // dlq get deleted on consumption.
+          LOG.debug("Retryable error occurred while processing an event: {}", ex.getMessage());
           // If cause is null, then it is a plain Spanner exception
-          outputWithTag(c, Constants.RETRYABLE_ERROR_TAG, ex.getMessage(), spannerRec);
+          outputWithTag(c, Constants.RETRYABLE_ERROR_TAG, message, spannerRec);
         } else if (cause instanceof InvalidTransformationException) {
           invalidTransformationException.inc();
-          outputWithTag(c, Constants.PERMANENT_ERROR_TAG, cause.getMessage(), spannerRec);
+          outputWithTag(c, Constants.PERMANENT_ERROR_TAG, message, spannerRec);
         } else if (cause instanceof ChangeEventConvertorException
-            || cause instanceof CodecNotFoundException) {
-          outputWithTag(c, Constants.PERMANENT_ERROR_TAG, cause.getMessage(), spannerRec);
-        } else if (cause instanceof IllegalStateException
-            || cause instanceof com.mysql.cj.jdbc.exceptions.CommunicationsException
-            || cause instanceof java.sql.SQLIntegrityConstraintViolationException
-            || cause instanceof java.sql.SQLTransientConnectionException
-            || cause instanceof ConnectionInitException
-            || cause instanceof DriverTimeoutException
-            || cause instanceof AllNodesFailedException
-            || cause instanceof BusyConnectionException
-            || cause instanceof NodeUnavailableException
-            || cause instanceof QueryExecutionException
-            || cause instanceof ConnectionException) {
-          outputWithTag(c, Constants.RETRYABLE_ERROR_TAG, cause.getMessage(), spannerRec);
-        } else if (cause instanceof java.sql.SQLNonTransientConnectionException) {
-          SQLNonTransientConnectionException e = (SQLNonTransientConnectionException) cause;
+            || cause instanceof CodecNotFoundException
+            || cause instanceof SQLSyntaxErrorException
+            || cause instanceof SQLDataException) {
+          outputWithTag(c, Constants.PERMANENT_ERROR_TAG, message, spannerRec);
+        } else if (cause instanceof SQLNonTransientConnectionException e) {
           // https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html
           // error codes 1053,1161 and 1159 can be retried
           if (e.getErrorCode() == 1053 || e.getErrorCode() == 1159 || e.getErrorCode() == 1161) {
-            outputWithTag(c, Constants.RETRYABLE_ERROR_TAG, e.getMessage(), spannerRec);
+            LOG.debug("Retryable error occurred while processing an event: {}", message);
+            outputWithTag(c, Constants.RETRYABLE_ERROR_TAG, message, spannerRec);
```

Comment:
If everything is retriable "by default", why is there a special branch for this use-case?